backend/services/tasks/pokemon_tasks.py

```diff
@@ -10,28 +10,45 @@
 @celery_app.task
 def save_all_pokemons_from_pokeapi():
     pokemons = get_all_pokemons()
-
-    for pokemon in pokemons:
-        existing_pokemon = Pokemon.objects.filter(name=pokemon["name"]).first()
-
-        if existing_pokemon:
-            continue
-
-        new_pokemon = get_pokemon_by_name(pokemon["name"])
-        poke_id = new_pokemon["id"]
-        name = new_pokemon["name"]
-        attack = new_pokemon["stats"][1]["base_stat"]
-        defense = new_pokemon["stats"][2]["base_stat"]
-        hit_points = new_pokemon["stats"][0]["base_stat"]
-        image = new_pokemon["sprites"]["front_default"]
-
-        Pokemon.objects.create(
-            poke_id=poke_id,
-            name=name,
-            attack=attack,
-            defense=defense,
-            hit_points=hit_points,
-            image=image,
+    pokemon_names = set(p["name"] for p in pokemons)
+    existing_pokemon_names = set(
+        Pokemon.objects.filter(name__in=pokemon_names).values_list("name", flat=True)
+    )
+    missing_pokemon_names = pokemon_names - existing_pokemon_names
+
+    pokemons_to_be_created = []
+    for name in missing_pokemon_names:
+        new_pokemon = get_pokemon_by_name(name)
+        pokemons_to_be_created.append(
+            Pokemon(
+                poke_id=new_pokemon["id"],
+                name=new_pokemon["name"],
+                attack=new_pokemon["stats"][1]["base_stat"],
+                defense=new_pokemon["stats"][2]["base_stat"],
+                hit_points=new_pokemon["stats"][0]["base_stat"],
+                image=new_pokemon["sprites"]["front_default"],
+            )
+        )
+    Pokemon.objects.bulk_create(pokemons_to_be_created)
+
+    existing_pokemons = Pokemon.objects.filter(name__in=existing_pokemon_names)
```

Comment:
Now you must run this BEFORE the bulk_create, otherwise this filter will include the pokemons you've just created.

Reply:
Also, you already have `existing_pokemon_names`, so maybe you don't even need this query?

Reply:
`bulk_update` requires the primary key of the objects.